### PR TITLE
Fix for processing border shorthand in filter transformation

### LINKED ISSUE #1010: "border-color" gets removed
## Are you reporting a feature request or a bug?

Bug

## Provide detailed reproduction steps (if any)

1. Set ACF to `config.extraAllowedContent = 'td{border-color,border-style,border-width}';`.
2. Open `replacebycode.html` sample in IE11.
3. Switch to source mode and paste:
```
<table style="width:500px">
<tbody>
<tr>
<td style="border-color:red; border-style:solid; border-width:2pt">&nbsp;</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
```
4. Switch to wysiwyg mode.

### Expected result

Broder in first cell is red.

### Actual result

Broder in first cell isn't red and when inspecting HTML with dev-tools, you can see that `border-color:red;` gets removed.

## Other details

* Browser: IE11
* OS: Windoes
* CKEditor version: 4.6+
* Installed CKEditor plugins: table, tabletools

Problem can be reproduced in IE11 from CKEditor 4.6.0.

### PR DESCRIPTION
**Pull request by msamsel**
## What is the purpose of this pull request?

Bug fix

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What changes did you make?
- change filter transformation `splitBorderShorthand` to properly split the border style and use tools method instead of having own logic,
- there exist an issue with processing border color with space which is an upstream here: #1356 

close #1010

### CODE REVIEW

File: tests/core/filter/manual/bordershorthandtransformation.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+@bender-tags: bug, 1010, 4.8.1, bug
+@bender-ui: collapsed
+@bender-ckeditor-plugins: wysiwygarea, toolbar, sourcearea, table, tabletools, contextmenu, dialogadvtab
+
+----
+
+1. Copy-paste table to editor.
+2. Switch between sourcemode and wysiwyg.
+3. Check TD element's style
```
Comment (f1ames) [inline tests/core/filter/manual/bordershorthandtransformation.md]: This scenario is a little confusing IMHO. The first step says to copy/paste table to editor, but there is no table, there is only table HTML.
Maybe it should say something like _Copy table HTML. Paste into editor in source mode._ and then the rest of scenario with switching and checking color. WDYT?

File: tests/core/filter/manual/bordershorthandtransformation.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+@bender-tags: bug, 1010, 4.8.1, bug
```
Comment (f1ames) [inline tests/core/filter/manual/bordershorthandtransformation.md]: Duplicated `bug` tag.

File: tests/core/filter/manual/bordershorthandtransformation.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+@bender-ui: collapsed
+@bender-ckeditor-plugins: wysiwygarea, toolbar, sourcearea, table, tabletools, contextmenu, dialogadvtab
+
+----
+
+1. Copy-paste table to editor.
+2. Switch between sourcemode and wysiwyg.
```
Comment (f1ames) [inline tests/core/filter/manual/bordershorthandtransformation.md]: Missing space in `sourcemode`, should be `source mode`.

File: tests/core/filter/manual/bordershorthandtransformation.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+		<tr>
+			<td style="border:2pt solid red;">&nbsp;</td>
+			<td>&nbsp;</td>
+		</tr>
+	</tbody>
+</table>
+</textarea>
+<script>
+	var editor = CKEDITOR.replace( 'editor1', {
+		extraAllowedContent: 'td{border,border-*}',
```
Comment (f1ames) [inline tests/core/filter/manual/bordershorthandtransformation.html]: Unnecessary comma on the end of this line.

**Comment by f1ames** [comment]
As for IE11, this is an known issue, see #473.

**Comment by msamsel** [comment]
I ignore test for IE8 as its problem is related to this case: #1396,
I replace border percentage value with allowed one, what breaks tests for IE9 and IE10.
I've made additional beautified assertion, which also detects and corrects IE11 upstream issue.

**Comment by msamsel** [comment]
In most cases of PFW test errors, there appeared border-color, which previously wasn't present. Also from some unknown reason span styles order was replaced in few PFW tests.

**Pull request merged**